### PR TITLE
Use more specific PHPDoc

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is a patch.

## Changelog

```markdown
### Added
- Add more `@method` annotation to propagate new methods of `CategoryManagerInterface`

### Removed
- Remove superfluous PHPDoc 

### Fixed
- Match PHPDoc with doctrine model 
```

### CODE REVIEW

File: src/Entity/CategoryManager.php [hunk 1]
```diff
@@ -155,13 +120,6 @@ public function getRootCategoriesForContext(ContextInterface $context = null)
         return $this->categories[$context->getId()];
     }
 
-    /**
-     * @todo NEXT_MAJOR: remove this method.
```
Comment (OskarStark) [inline src/Entity/CategoryManager.php]: Shouldn’t we keep this?

Comment (core23) [reply src/Entity/CategoryManager.php]: IMHO we don't need this. We just need to port all `@method` methods

File: src/Entity/CategoryManager.php [hunk 2]
```diff
@@ -48,12 +47,6 @@ public function __construct($class, ManagerRegistry $registry, ContextManagerInt
 
     /**
      * Returns a pager to iterate over the root category.
-     *
-     * @param int   $page
```
Comment (OskarStark) [inline src/Entity/CategoryManager.php]: Why removing this? There is no typehint so this is the only place to get the type, isn’t it?

File: src/Entity/CategoryManager.php (hunk 1 above)
Comment (OskarStark) [reply src/Entity/CategoryManager.php]: For adding yes, but this is about removing a method

Comment (core23) [reply src/Entity/CategoryManager.php]: Oh, you're right

File: src/Entity/CategoryManager.php (hunk 2 above)
Comment (core23) [reply src/Entity/CategoryManager.php]: The type hint was moved: https://github.com/sonata-project/SonataClassificationBundle/pull/463/commits/e5739d249bf334ddb3f49c6df79c804d12f2dd10#diff-4d9dd8b669cb2a016d76d86e7d6f0384R20

File: src/Entity/CategoryManager.php (hunk 1 above)
Comment (OskarStark) [reply src/Entity/CategoryManager.php]: There are others too IIRC

Comment (core23) [reply src/Entity/CategoryManager.php]: Fixed

**Pull request merged**